lib/ansible/modules/mount_facts.py

```diff
@@ -0,0 +1,637 @@
+                    module.warn(f"Ignoring repeat mounts for {mount_point}. "
+                                "You can disable this warning by configuring the 'include_aggregate_mounts' option as True or False.")
+            yield mount
+
+
+def get_mount_facts(module: AnsibleModule):
+    """List and filter mounts, returning a dictionary containing mount points as the keys (last listed wins)."""
+    seconds = module.params["timeout"]
+
+    # merge sources based on the mount point (first definition + source wins)
+    mount_points: dict[str, list[dict[str, str | int | dict[str, str]]]] = {}
+    aggregate_mounts: list[dict[str, str | int | dict[str, str]]] = []
+    for source, mount, origin, fields in gen_mounts_by_source(module):
+        device = fields["device"]
+        fstype = fields["fstype"]
+
+        # Convert UUIDs in Linux /etc/fstab to device paths
+        # TODO need similar for OpenBSD which lists UUIDS (without the UUID= prefix) in /etc/fstab, needs another approach though.
+        uuid = None
+        if device.startswith("UUID="):
+            uuid = device.split("=", 1)[1]
+            device = get_device_by_uuid(module, uuid) or device
+
+        if not any(fnmatch(device, pattern) for pattern in module.params["devices"] or ["*"]):
+            continue
+        if not any(fnmatch(fstype, pattern) for pattern in module.params["fstypes"] or ["*"]):
+            continue
+
+        timed_func = _timeout.timeout(seconds, f"Timed out getting mount size for mount {mount} (type {fstype})")(get_mount_size)
+        if mount_size := handle_timeout(module)(timed_func)(mount):
+            fields.update(mount_size)
+
+        if uuid is None:
+            with suppress(subprocess.CalledProcessError):
+                uuid = get_partition_uuid(module, device)
+
+        fields.update({"ansible_context": {"source": source, "source_data": origin}, "uuid": uuid})
+        if mount not in mount_points:
+            mount_points[mount] = fields
+        aggregate_mounts.append(fields)
+
+    return mount_points, aggregate_mounts
+
+
+def get_argument_spec():
+    """Helper returning the argument spec."""
+    return dict(
+        sources=dict(type="list", elements="str", default=None),
+        mount_binary=dict(default="mount", type="raw"),
+        devices=dict(type="list", elements="str", default=None),
+        fstypes=dict(type="list", elements="str", default=None),
+        timeout=dict(type="float"),
+        on_timeout=dict(choices=["error", "warn", "ignore"], default="error"),
+        include_aggregate_mounts=dict(default=None, type="bool"),
+    )
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=get_argument_spec(),
+        supports_check_mode=True,
+    )
+    if (seconds := module.params["timeout"]) is not None and seconds <= 0:
+        module.fail_json(msg="argument 'timeout' must be a positive integer or null, not {seconds}")
```

Comment:
(since the argspec says `float`):
```suggestion
        module.fail_json(msg="argument 'timeout' must be a positive number or null, not {seconds}")
```